sdk/identity/azure-identity/azure/identity/_internal/client_secret_credential_base.py

```diff
@@ -27,7 +31,15 @@ def __init__(self, tenant_id, client_id, client_secret, **kwargs):
                 "tenant_id should be an Azure Active Directory tenant's id (also called its 'directory id')"
             )
 
-        self._client = self._get_auth_client(tenant_id, client_id, **kwargs)
+        enable_persistent_cache = kwargs.pop("enable_persistent_cache", False)
+        if enable_persistent_cache:
+            allow_unencrypted = kwargs.pop("allow_unencrypted_cache", False)
+            cache = load_service_principal_cache(allow_unencrypted)
+        else:
+            cache = TokenCache()
```

Comment:
I see in AadClientBase, we have self._cache = cache or TokenCache().

So why not just pass in None?

Reply:
To spare a future reader of this line from wondering what `cache = None` means.